dice.py

```diff
@@ -1,23 +1,88 @@
+* The number of dice and how many faces they have is currently hardcoded under configurables
+* The return_dice_as_text function returns "a 11" instead of "an 11". I really CBA to fix that
+"""
+
 import random
 
-repeat = True
-while repeat:
-    no = random.randint(1, 6)
-    print("You rolled", no)
-    if no == 1:
-        print("You feel somewhat disappointed")
-    if no == 2:
-        print("You look at the two dots and begin to mourn a lover lost at sea")
-    if no == 3:
-        print("Three is the magic number, you turn into a frog")
-    if no == 4:
-        print("4! You win...not really, you're just rolling a dice. Get some mates!")
-    if no == 5:
-        print("You bleed a little")
-    if no == 6:
-        print("You begin to wonder what this point of this is")
-    if no == 7:
-        print("err...what? This is literally impossible")
-
-    print("Do you want to roll again Y/N (please roll again, it...it pleases me?)")
-    repeat = "Y" in input()
\ No newline at end of file
+"""Configurables"""
+number_of_dice = 3
+faces = 6
+
+
+def main():
+    repeat = True
+    while repeat:
+        display(roll(number_of_dice, faces))
+        print("Do you want to roll again Y/N (please roll again, it...it pleases me?)")
+        repeat = "Y" in input().upper()
+
+
+def roll(number_of_dice, faces):
+    return [random.randint(1, faces) for n in range(number_of_dice)]
+
+
+def display(dice):
+    if all(number <= 6 for number in dice):
+        for number in dice:
+            if number == 1:
+                print(" -------")
+                print("｜     ｜")
+                print("｜  o  ｜")
+                print("｜     ｜")
+                print(" -------")
+            elif number == 2:
+                print(" -------")
+                print("｜o    ｜")
+                print("｜     ｜")
+                print("｜    o｜")
+                print(" -------")
+            elif number == 3:
+                print(" -------")
+                print("｜o    ｜")
+                print("｜  o  ｜")
+                print("｜    o｜")
+                print(" -------")
+            elif number == 4:
+                print(" -------")
+                print("｜o   o｜")
+                print("｜     ｜")
+                print("｜o   o｜")
+                print(" -------")
+            elif number == 5:
+                print(" -------")
+                print("｜o   o｜")
+                print("｜  o  ｜")
+                print("｜o   o｜")
+                print(" -------")
+            elif number == 6:
+                print(" -------")
+                print("｜o   o｜")
+                print("｜o   o｜")
+                print("｜o   o｜")
+                print(" -------")
+    print("You rolled " + return_dice_as_text(dice))
+
+
+def return_dice_as_text(dice):
+    output = ""
+    for i in range(len(dice)):
+        if i == 0:
+            output += f"a {dice[i]}"
+            if len(dice) == 1:
+                break
```

Comment:
Does 'break' create a new line or is it more a break in the sequence or something?

Reply:
Break stops the current loop. Here it is to check if we're on the last item in the list **and** the list is exactly one item long. Usually we would need 'and a' before the last item in the list but not if the list only has one thing in it. Like I said, this whole function might be a bit too complicated but I got carried away.